tests/trx_generator/trx_generator_tests.cpp

```diff
@@ -1,10 +1,58 @@
 #include <trx_provider.hpp>
-#include <trx_generator.cpp>
+#include <trx_generator.hpp>
+#include <http_client_async.hpp>
+#include <simple_rest_server.hpp>
+
 #define BOOST_TEST_MODULE trx_generator_tests
 #include <boost/test/included/unit_test.hpp>
 
+using namespace eosio;
 using namespace eosio::testing;
 
+static const char* api_name = "/v1/chain/test";
+
+namespace http = boost::beast::http;
+struct trx_gen_impl : rest::simple_server<trx_gen_impl> {
+
+   std::string server_header() const { return "/"; }
+
+   void log_error(char const* what, const std::string& message) {
+      elog("${what}: ${message}", ("what", what)("message", message));
+   }
+
+   bool allow_method(http::verb method) const { return method == http::verb::post; }
+
+   std::optional<http::response<http::string_body>> on_request(http::request<http::string_body>&& req) {
+      if (req.target() != api_name)
+         return {};
+      http::response<http::string_body> res{http::status::ok, req.version()};
+      // Respond to POST request
+      res.set(http::field::server, server_header());
+      res.set(http::field::content_type, "text/plain");
+      res.keep_alive(req.keep_alive());
+      res.body() = req.body();
```

Comment:
Add comment that this echos back the request. I would name it `echo_server`.

Reply:
Addressed: [f02287e](https://github.com/AntelopeIO/leap/pull/1052/commits/f02287e44535e51b7ae21d1c90db73c40f51d644)